charts/shoot-core/components/charts/nodelocal-dns/templates/daemonset.yaml

```diff
@@ -0,0 +1,102 @@
+---
+apiVersion: {{ include "daemonsetversion" . }}
+kind: DaemonSet
+metadata:
+  name: node-local-dns
+  namespace: kube-system
+  labels:
+    gardener.cloud/role: system-component
+    garden.sapcloud.io/role: system-component
+    origin: gardener
+    k8s-app: node-local-dns
+spec:
+  updateStrategy:
+    rollingUpdate:
+      maxUnavailable: 10%
+  selector:
+    matchLabels:
+      k8s-app: node-local-dns
+  template:
+    metadata:
+      labels:
+        k8s-app: node-local-dns
+        networking.gardener.cloud/to-dns: allowed
+        # we won't be using the checksum of the configmap since coreDNS provides the "reload" plugins that does the reload if config changes.
+      annotations:
+        prometheus.io/port: "{{ .Values.prometheus.port }}"
+        prometheus.io/scrape: "{{ .Values.prometheus.scrape }}"
```

Comment:
Hmm, I don't see any node-local-dns scrape targets in prometheus after enabling the feature gate.

Reply:
I removed monitoring from this PR.